redbot/cogs/mutes/mutes.py

```diff
@@ -0,0 +1,1212 @@
+    async def mute_role(self, ctx: commands.Context, *, role: discord.Role = None):
+        """Sets the role to be applied when muting a user.
+
+        If no role is setup the bot will attempt to mute a user by setting
+        channel overwrites in all channels to prevent the user from sending messages.
+        """
+        if not role:
+            await self.config.guild(ctx.guild).mute_role.set(None)
+            await self.config.guild(ctx.guild).sent_instructions(False)
+            # reset this to warn users next time they may have accidentally
+            # removed the mute role
+            await ctx.send(_("Channel overwrites will be used for mutes instead."))
+        else:
+            await self.config.guild(ctx.guild).mute_role.set(role.id)
+            await ctx.send(_("Mute role set to {role}").format(role=role.name))
+
+    @muteset.command(name="makerole")
+    @checks.bot_has_permissions(manage_roles=True)
+    async def make_mute_role(self, ctx: commands.Context, *, name: str):
+        """Create a Muted role.
+
+        This will create a role and apply overwrites to all available channels
+        to more easily setup muting a user.
+
+        If you already have a muted role created on the server use
+        `[p]muteset role ROLE_NAME_HERE`
+        """
+        async with ctx.typing():
+            perms = discord.Permissions()
+            perms.update(send_messages=False, speak=False, add_reactions=False)
+            try:
+                role = await ctx.guild.create_role(
+                    name=name, permissions=perms, reason=_("Mute role setup")
+                )
+            except discord.errors.Forbidden:
+                return await ctx.send(_("I could not create a muted role in this server."))
+            errors = []
+            tasks = []
+            for channel in ctx.guild.channels:
+                tasks.append(self._set_mute_role_overwrites(role, channel))
+            errors = await asyncio.gather(*tasks)
+            if any(errors):
+                msg = _(
+                    "I could not set overwrites for the following channels: {channels}"
+                ).format(channels=humanize_list([i for i in errors if i]))
+                for page in pagify(msg):
+                    await ctx.send(page)
+            await self.config.guild(ctx.guild).mute_role.set(role.id)
+            await ctx.send(_("Mute role set to {role}").format(role=role.name))
+
+    async def _set_mute_role_overwrites(
+        self, role: discord.Role, channel: discord.abc.GuildChannel
+    ) -> Optional[str]:
+        """
+        This sets the supplied role and channel overwrites to what we want
+        by default for a mute role
+        """
+        overs = discord.PermissionOverwrite()
+        if isinstance(channel, discord.TextChannel):
+            overs.send_messages = False
+            overs.add_reactions = False
+        if isinstance(channel, discord.VoiceChannel):
+            overs.speak = False
```

Comment:
I think it would be nice if we also applied these to category channels so that creating a new channel in the category automatically makes the appropriate channel overwrite. It is not required, I just think it would be nice.

Here's a code suggestion which works just fine as it is not required to only specify the overwrites that are actually related to the channel type:
```suggestion
        overs.send_messages = False
        overs.add_reactions = False
        overs.speak = False
```